tests/Measurement.test.ts

```diff
@@ -0,0 +1,150 @@
+import Measurement from '../src/Measurement'
+import Vector from '../src/Vector'
+import Dimension from '../src/Dimension'
+import { Cx } from '../src/Complex'
+import './customMatchers'
+
+describe('Measurement', () => {
+  it('the simplest measurement', () => {
+    const vector = Vector.fromSparseCoordNames(
+      [
+        ['0', Cx(1)],
+        ['1', Cx(1)],
+      ],
+      [Dimension.qubit()],
+    ).normalize()
+    const m = new Measurement([{ name: [], vector }])
+    expect(m.states.length).toBe(1)
+    expect(m.toString()).toBe('100.0% [] 0.71 exp(0.00τi) |0⟩ + 0.71 exp(0.00τi) |1⟩')
+    const projections = [
+      { name: ['zero'], vector: Vector.indicator([Dimension.qubit()], '0') },
+      { name: ['one'], vector: Vector.indicator([Dimension.qubit()], '1') },
+    ]
+    const newM = m.projectiveMeasurement([0], projections)
+    expect(newM.toString()).toBe(['50.0% [zero] 1.00 exp(0.00τi) |⟩', '50.0% [one] 1.00 exp(0.00τi) |⟩'].join('\n'))
+  })
+
+  it('two particles', () => {
+    const vector = Vector.fromSparseCoordNames(
+      [
+        ['01', Cx(1)],
+        ['10', Cx(-1)],
+      ],
+      [Dimension.qubit(), Dimension.qubit()],
+    ).normalize()
+    const m = new Measurement([{ name: [], vector }])
+    const projections = [
+      { name: ['zero'], vector: Vector.indicator([Dimension.qubit()], '0') },
```

Comment:
Same argument as with `fromVector`, i feel like the `name` parameter here should not be an array, but a single value. So it becomes
```ts
{ name: 'zero', vector: Vector.indicator([Dimension.qubit()], '0') },
```

Or we could even enforce unique names in the projections by using keyed objects, making the syntax nicer to use in the process.
```ts
const projections = {
	zero: Vector.indicator([Dimension.qubit()], '0'),
	one: Vector.indicator([Dimension.qubit()], '1'),
}
```

Is there any usecase this would prevent?

Reply:
This is "by design". For example, for detector we can have two outcomes named "detected" and 6 "hit a wall". 

At the same time, I understand that other pattern (unique names, collapsed only later) might be a choice that also works.